freqtrade/freqai/freqai_interface.py

```diff
@@ -505,42 +507,47 @@ def check_if_feature_list_matches_strategy(
                 "feature_engineering_* functions"
             )
 
-    def define_data_pipeline(self, dk: FreqaiDataKitchen) -> None:
+    def define_data_pipeline(self) -> Pipeline:
         ft_params = self.freqai_info["feature_parameters"]
-        dk.feature_pipeline = Pipeline([
-            ('const', ds.DataSieveVarianceThreshold(threshold=0)),
-            ('scaler', ds.DataSieveMinMaxScaler(feature_range=(-1, 1)))
+        feature_pipeline = Pipeline([
+            ('const', ds.VarianceThreshold(threshold=0)),
+            ('scaler', SKLearnWrapper(MinMaxScaler(feature_range=(-1, 1))))
             ])
 
         if ft_params.get("principal_component_analysis", False):
-            dk.feature_pipeline.steps += [('pca', ds.DataSievePCA())]
-            dk.feature_pipeline.steps += [('post-pca-scaler',
-                                           ds.DataSieveMinMaxScaler(feature_range=(-1, 1)))]
+            feature_pipeline.append(('pca', ds.PCA()))
+            feature_pipeline.append(('post-pca-scaler',
+                                     SKLearnWrapper(MinMaxScaler(feature_range=(-1, 1)))))
 
         if ft_params.get("use_SVM_to_remove_outliers", False):
             svm_params = ft_params.get(
                 "svm_params", {"shuffle": False, "nu": 0.01})
-            dk.feature_pipeline.steps += [('svm', ds.SVMOutlierExtractor(**svm_params))]
+            feature_pipeline.append(('svm', ds.SVMOutlierExtractor(**svm_params)))
 
         di = ft_params.get("DI_threshold", 0)
         if di:
-            dk.feature_pipeline.steps += [('di', ds.DissimilarityIndex(di_threshold=di))]
+            feature_pipeline.append(('di', ds.DissimilarityIndex(di_threshold=di)))
 
         if ft_params.get("use_DBSCAN_to_remove_outliers", False):
-            dk.feature_pipeline.steps += [('dbscan', ds.DataSieveDBSCAN())]
+            feature_pipeline.append(('dbscan', ds.DBSCAN()))
 
-        dk.feature_pipeline.fitparams = dk.feature_pipeline._validate_fitparams(
-            {}, dk.feature_pipeline.steps)
+        sigma = self.freqai_info["feature_parameters"].get('noise_standard_deviation', 0)
+        if sigma:
+            feature_pipeline.append(('noise', ds.Noise(sigma=sigma)))
 
-        # if self.freqai_info["feature_parameters"].get('noise_standard_deviation', 0):
-        #     dk.pipeline.extend(('noise', ds.Noise()))
+        feature_pipeline.fitparams = feature_pipeline._validate_fitparams(
```

Comment:
is this a mandatory call a user is expected to make? 
if yes, why is the method "private" (_ prefixed methods are considered private by convention).
Also if yes - should this call be abstracted away from the user (by moving it to "after" the calls)?

Also, is there some further docs to datasieve? i've looked at the readme (couldn't find other docs) - which doesn't seem to mention this method at all ?

Reply:
I think then this should be done differently here, too.
there's nothing preventing us from creating a list we append the single entries to - and calling `Pipeline(pipelinesteps)` at the end of the function.

It's a private, undocumented method on an external (code is not in this repo) module - so it shouldn't be called directly (otherwise it should be made public).

Reply:
You are right, easier to build the list and then instantiate the object at once at the end. Done.